From 380c1ef8393628c3e5098518912ad71f9ff08c65 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 21 Apr 2022 11:36:35 +0800
Subject: [PATCH 16/36] HACK: arm64: dts: rockchip: disable usb3 on quartz64

USB3 on Quartz64 is of bad quality because of sharing lines with SATA.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
Ref: Contact the author.
---
 arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
index 854d02b46e6f..2de152da3fac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-quartz64-a.dts
@@ -788,6 +788,10 @@ &usb_host0_xhci {
 
 /* usb3 controller is muxed with sata1 */
 &usb_host1_xhci {
+	phys = <&usb2phy1_otg>;
+	phy-names = "usb2-phy";
+	extcon = <&usb2phy1>;
+	maximum-speed = "high-speed";
 	status = "okay";
 };
 
@@ -801,7 +805,7 @@ &usb2phy0_host {
 };
 
 &usb2phy0_otg {
-	phy-supply = <&vcc5v0_usb20_otg>;
+	phy-supply = <&vcc5v0_usb20_host>;
 	status = "okay";
 };
 
-- 
2.46.0

